src/ui/common/src/components/resources/dialogs/onDemandKubernetesDialog.tsx

```diff
@@ -283,9 +299,94 @@ const OnDemandK8sAWSStep: React.FC<ResourceDialogProps> = ({
   );
 };
 
+const OnDemandK8sGCPStep: React.FC<ResourceDialogProps> = ({
+  user,
+  editMode,
+  onCloseDialog,
+  loading,
+  disabled,
+}) => {
+  const methods = useFormContext();
+  const dispatch: AppDispatch = useDispatch();
+
+  return (
+    <>
+      <DialogTitle sx={{ display: 'flex', alignItems: 'center' }}>
+        <ResourceLogo
+          service={'Aqueduct'}
+          activated={SupportedResources['Aqueduct'].activated}
+          size="small"
+        />
+        <div>
+          <Typography variant="h5" sx={{ color: 'black' }}>
+            +
+          </Typography>
+        </div>
+        <ResourceLogo
+          service={'AWS'}
+          activated={SupportedResources['Kubernetes'].activated}
+          size="small"
+        />
+        <div>
+          <Typography variant="h5" sx={{ color: 'black' }}>
+            Aqueduct-managed Kubernetes on GCP
+          </Typography>
+        </div>
+      </DialogTitle>
+      <ResourceTextInputField
+        name="name"
+        spellCheck={false}
+        required={true}
+        label="Name*"
+        description="Provide a unique name to refer to this resource."
+        placeholder={'my_kubernetes_resource'}
+        onChange={(event) => {
+          methods.setValue('name', event.target.value);
+        }}
+        disabled={false}
+      />
+      <GCPDialog
+        user={user}
+        disabled={disabled}
+        loading={loading}
+        onCloseDialog={onCloseDialog}
+        editMode={false}
+      />
+      <DialogActionButtons
+        onCloseDialog={onCloseDialog}
+        loading={loading}
+        disabled={disabled}
+        onSubmit={async () => {
+          await methods.handleSubmit((data) => {
+            // Remove the name field from request body to avoid pydantic errors.
+            // Name needs to be passed in as a header instead. Dunno why it's not part of the body :shrug:
+            const name = data.name;
+            delete data.name;
+            // Remove extraneous fields if they are added when filling out the form.
+            delete data.k8s_type;
+            delete data.type;
+
+            data.cloud_provider = 'GCP';
+
+            dispatch(
+              handleConnectToNewResource({
+                apiKey: user.apiKey,
+                service: 'Kubernetes',
+                name: name,
+                config: data,
+              })
+            );
+          })(); // Remember the last two parens to call the function!
+        }}
+      />
+    </>
+  );
+};
+
 interface OnDemandK8sStepProps extends ResourceDialogProps {
   handlePrevious: () => void;
   handleAWSClick: () => void;
+  handleGCPClick: () => void;
```

Comment:
We've changed this to `onSelectProvider` so we shouldn't need to add an extra handler here